draft-ietf-taps-arch.md

```diff
@@ -216,35 +216,53 @@ Transport Services implementations are meant to be flexible at connection establ
 
-## Common APIs for Common Features
+In order to provide this functionality without causing incompatibility or introducing security vulnerabilities, Transport Services APIs and Implementations should follow the set of requirements described here.
 
-Functionality that is common across multiple transport protocols ought to be accessible through a unified set of API calls. An application using a Transport Services API can implement logic for its basic use of transport networking (establishing the transport, and sending and receiving data) once, and expect that implementation to continue to function as the transports change.
+## Provide Common APIs for Common Features
 
-As a baseline, any Transport Services API needs to allow access to the distilled minimal set of features offered by transport protocols {{?I-D.ietf-taps-minset}}.
+Any functionality that is common across multiple transport protocols SHOULD be made accessible through a unified set of Transport Services API calls. As a baseline, any Transport Services API MUST allow access to the minimal set of features offered by transport protocols {{?I-D.ietf-taps-minset}}.
 
-## Access to Specialized Features
+An application can specify constraints and preferences for the protocols, features, and network interfaces it will use via Properties. A Transport Services API SHOULD offer Properties that are common to multiple transport protocols, in order to enable the system to appropriately select between protocols that offer equivalent features. Similarly, a Transport Services API SHOULD offer Properties that are applicable to a variety network layer interfaces and paths, in order to permit racing of different network paths without affecting the applications using the system. Each Property is expected to have a default value.
 
-There are applications that will need to control fine-grained details of transport protocols to optimize their behavior and ensure compatibility with remote systems. A Transport Services system therefore ought to also permit more specialized protocol features to be used. The interface for these specialized options ought to be exposed differently from the common options to ensure flexibility.
+Properties defaults SHOULD be selected to ensure correctness for the widest set of applications, while providing the widest set of options for selection. For example, since both applications that require data reliability and those that do not require data reliability can function correctly when a protocol provides reliability, reliability ought to be required by default. In contrast, the default value for a Property regarding the selection of network interfaces ought to permit as many interfaces as possible.
```

Comment:
I'd suggest: "The default for Properties". 
I'd suggest: "The default for this property is to require reliability". 
.... because we need to be careful in how we use the word 'require' in TAPS

Reply:
Updated